index.js

```diff
@@ -18,7 +18,7 @@ module.exports =
  */
 function resolveBinding(name) {
   if (debugging ) return process.binding(name.replace(/\.node$/,''));
-  return bindings({ bindings: name, module_root: bindings.getRoot(module.parent) });
+  return bindings({ bindings: name, module_root: bindings.getRoot(module.parent.id) });
```

Comment:
This works in some cases (so in those it must have been a string). Those would break now though since they have no `.id` property.
I might be wrong about this (it was a while ago).
If you aren't sure either could we do:

```js
var id = module.parent.id || module.parent;
return bindings({ bindings: name, module_root: bindings.getRoot(id) });
```
?